tests/unit/operators/test_datarobot.py

```diff
@@ -103,31 +105,93 @@ def test_operator_deploy_recommended_model(mocker):
     )
 
 
-def test_operator_score_predictions(mocker):
-    job_mock = mocker.Mock()
-    job_mock.id = "job-id"
-    score_mock = mocker.patch.object(dr.BatchPredictionJob, "score", return_value=job_mock)
-
-    operator = ScorePredictionsOperator(task_id="score_predictions", deployment_id="deployment-id")
-    settings = {
-        "intake_settings": {
+@pytest.fixture(
+    scope="module",
+    params=[
+        {
+            "type": "dataset",
+            "datasetId": "dataset-id",
+        },
+        {
             "type": "s3",
             "url": "s3://path/to/scoring_dataset.csv",
             "credential_id": "credential-id",
         },
+    ],
+)
+def score_settings(request):
+    return {
+        "intake_settings": request.param,
         "output_settings": {
             "type": "s3",
             "url": "s3://path/to/predictions.csv",
             "credential_id": "credential-id",
         },
     }
+
+
+def test_operator_score_predictions(mocker, score_settings):
+    job_id = "job-id"
+
+    # Mocks out the response from the BatchPredictionJob POST request
+    # rather than mocking the whole `score` method
+    post_mock = mocker.Mock()
+    post_mock.json = lambda: job_id
+    post_mock.headers = {
+        "Location": f"https://app.datarobot.com/api/v2/BatchPredictionJob/{job_id}/"
+    }
+    mocker.patch.object(dr.BatchPredictionJob._client, "post", return_value=post_mock)
+
+    job_mock = mocker.Mock()
+    job_mock.id = job_id
+    score_mock = mocker.patch.object(
+        dr.BatchPredictionJob, "get", return_value=job_mock
+    )
+
+    if score_settings["intake_settings"]["type"] == "dataset":
+        dataset_mock = mocker.Mock()
+        dataset_mock.id = "dataset-id"
+        dataset_mock.__class__ = dr.models.dataset.Dataset
+        mocker.patch.object(dr.Dataset, "get", return_value=dataset_mock)
+
+    operator = ScorePredictionsOperator(
+        task_id="score_predictions", deployment_id="deployment-id"
+    )
+
     job_id = operator.execute(
         context={
             "params": {
-                "score_settings": settings,
+                "score_settings": score_settings,
             }
         }
     )
 
     assert job_id == "job-id"
-    score_mock.assert_called_with("deployment-id", **settings)
+
+    score_mock.assert_called_with(job_id)
```

Comment:
I think it's important to check that `score_settings` contains a dataset object with the ID from the user-supplied config. Could you make sure it's tested? I'm ok with mocking `.score()` method and checking that it gets the right params supplied. We should not worry about testing our Python SDK here, we should only test this integration library.